causalpy/skl_experiments.py

```diff
@@ -11,721 +11,64 @@
-        self.outcome_variable_name = y.design_info.column_names[0]
-
-        # TODO: `treated` is a deterministic function of x and treatment_threshold, so
-        # this could be a function rather than supplied data
-
-        # fit the model to all the data
-        self.model.fit(X=self.X, y=self.y)
-
-        # score the goodness of fit to all data
-        self.score = self.model.score(X=self.X, y=self.y)
 
-        # get the model predictions of the observed data
-        if self.bandwidth is not None:
-            xi = np.linspace(fmin, fmax, 200)
-        else:
-            xi = np.linspace(
-                np.min(self.data[self.running_variable_name]),
-                np.max(self.data[self.running_variable_name]),
-                200,
-            )
-        self.x_pred = pd.DataFrame(
-            {self.running_variable_name: xi, "treated": self._is_treated(xi)}
-        )
-        (new_x,) = build_design_matrices([self._x_design_info], self.x_pred)
-        self.pred = self.model.predict(X=np.asarray(new_x))
+# Ensure deprecation warnings are always shown in Jupyter Notebooks
+warnings.simplefilter("always", DeprecationWarning)
 
-        # calculate discontinuity by evaluating the difference in model expectation on
-        # either side of the discontinuity
-        # NOTE: `"treated": np.array([0, 1])`` assumes treatment is applied above
-        # (not below) the threshold
-        self.x_discon = pd.DataFrame(
-            {
-                self.running_variable_name: np.array(
-                    [
-                        self.treatment_threshold - self.epsilon,
-                        self.treatment_threshold + self.epsilon,
-                    ]
-                ),
-                "treated": np.array([0, 1]),
-            }
-        )
-        (new_x,) = build_design_matrices([self._x_design_info], self.x_discon)
-        self.pred_discon = self.model.predict(X=np.asarray(new_x))
-        self.discontinuity_at_threshold = np.squeeze(self.pred_discon[1]) - np.squeeze(
-            self.pred_discon[0]
-        )
 
-    def _is_treated(self, x):
-        """Returns ``True`` if ``x`` is greater than or equal to the treatment
-        threshold.
+def SyntheticControl(*args, **kwargs):
+    warnings.warn(
+        """causalpy.skl_experiments.SyntheticControl is deprecated and will be removed in a future release. Please use causalpy.experiments.SyntheticControl instead.""",
+        DeprecationWarning,
+        stacklevel=2,
+    )
+    return NewSyntheticControl(*args, **kwargs)
 
-        .. warning::
 
-            Assumes treatment is given to those ABOVE the treatment threshold.
-        """
-        return np.greater_equal(x, self.treatment_threshold)
+def DifferenceInDifferences(*args, **kwargs):
+    warnings.warn(
+        """causalpy.skl_experiments.DifferenceInDifferences is deprecated and will be removed in a future release. Please use causalpy.experiments.DifferenceInDifferences instead.""",
```

Comment:
Same deprecation suggestion here

Reply:
Well spotted - I've updated for the scikit-learn experiments in https://github.com/pymc-labs/CausalPy/pull/381/commits/66f22aaf0a428132b02f298b6d13032f81214413